### PR TITLE
Documentation - First pass

### PR DESCRIPTION
**Pull request by cxr00**
Added descriptions to classes, key variables, and non-overridden methods;

replaced deprecated FileUtils methods by adding CharSet parameter

### CODE REVIEW

File: src/com/timoschwarzer/hkmodinstaller/data/ModBundle.java [hunk 1]
```diff
@@ -31,15 +38,18 @@ public ModBundle(String filename) throws Exception {
         load();
     }
 
+    /** Ensures and loads all elements of the ModBundle from the filename given */
     private void load() throws Exception {
+        // Pre-initialization and version validation
         ZipEntry manifest = zipFile.getEntry("mod.json");
 
-        JSONObject manifestObject = new JSONObject(IOUtils.toString(zipFile.getInputStream(manifest)));
+        JSONObject manifestObject = new JSONObject(IOUtils.toString(zipFile.getInputStream(manifest), Charset.defaultCharset()));
```
Comment (timoschwarzer) [inline src/com/timoschwarzer/hkmodinstaller/data/ModBundle.java]: I don't think using the default OS charset here is a good idea. Imagine one has created a modbundle on a UNIX machine (UTF-8 charset) and opens it on a Windows machine (Windows 1252).
I think using UTF-8 everywhere is a good solution the mod developers can live with.
So could you replace `Charset.defaultCharset()` with `StandardCharsets.UTF_8`, please? :smile:

File: src/com/timoschwarzer/hkmodinstaller/data/ModDatabase.java [hunk 2]
```diff
@@ -38,9 +47,10 @@ private ModDatabase() throws Exception {
         this.bundlesDirectory = this.dataDirectory + "/bundles";
         this.originalDirectory = this.dataDirectory + "/original";
 
+        // Load or create config file
         final File configFile = new File(this.dataDirectory + "/config.json");
         if (configFile.exists()) {
-            configObject = new JSONObject(FileUtils.readFileToString(configFile));
+            configObject = new JSONObject(FileUtils.readFileToString(configFile, Charset.defaultCharset()));
```
Comment (timoschwarzer) [inline src/com/timoschwarzer/hkmodinstaller/data/ModDatabase.java]: ...same here ;-)

File: src/com/timoschwarzer/hkmodinstaller/data/ModDatabase.java [hunk 3]
```diff
@@ -108,10 +135,20 @@ public void deleteBundle(ModBundle bundle) throws Exception {
         initializeBundles();
     }
 
+
+    /** Save the ModDatabase's current config */
     private void saveConfig() throws IOException {
-        FileUtils.write(new File(this.dataDirectory + "/config.json"), configObject.toString());
+        FileUtils.writeStringToFile(new File(this.dataDirectory + "/config.json"), configObject.toString(), Charset.defaultCharset());
```
Comment (timoschwarzer) [inline src/com/timoschwarzer/hkmodinstaller/data/ModDatabase.java]: ... aaaaand here :grin:

**Review by timoschwarzer — CHANGES_REQUESTED** [review]

**Comment by timoschwarzer** [comment]
Hi there!

Thank you a lot for doing the things I'm too lazy to do! :sweat_smile: 
Please see my comments and apply the suggestions or leave a comment on them.

**Comment by cxr00** [comment]
Sure thing! The reason I did used the default charset is because the old write method simply redirected to writeStringToFile with the default charset as an argument. But UTF-8 is definitely better. I'll fix that right up

**Comment by cxr00** [comment]
All done! I also pulled a couple unused imports just cause.

**Comment by timoschwarzer** [comment]
Thank you! :)

**Pull request merged**